src/AltinnCore/Common/Helpers/InstantiationHelper.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using Altinn.Platform.Storage.Models;
+using AltinnCore.ServiceLibrary.Enums;
+using AltinnCore.ServiceLibrary.Models;
+
+namespace Common.Helpers
+{
+    /// <summary>
+    /// Various helper methods for instantiation
+    /// </summary>
+    public static class InstantiationHelper
+    {
+        /// <summary>
+        /// Filters a list of parties based on an applications allowed party types.
+        /// </summary>
+        /// <param name="parties">The list of parties to be filtered</param>
+        /// <param name="partyTypesAllowed">The allowed party types</param>
+        /// <returns>A list with the filtered parties</returns>
+        public static List<Party> FilterPartiesByAllowedPartyTypes(List<Party> parties, PartyTypesAllowed partyTypesAllowed)
+        {
+            List<Party> allowed = new List<Party>();
+            if (parties == null || partyTypesAllowed == null)
+            {
+                return allowed;
+            }
+
+            parties.ForEach(party =>
+            {
+                if (IsPartyAllowedToInstantiate(party, partyTypesAllowed))
+                {
+                    allowed.Add(party);
+                }
+            });
+            return allowed;
+        }
+
+        /// <summary>
+        /// Checks if a party is allowed to initiate an application based on the applications AllowedPartyTypes
+        /// </summary>
+        /// <param name="party">The party to check</param>
+        /// <param name="partyTypesAllowed">The allowed party types</param>
+        /// <returns>True or false</returns>
+        public static bool IsPartyAllowedToInstantiate(Party party, PartyTypesAllowed partyTypesAllowed)
+        {
+            if (party == null)
+            {
+                return false;
+            }
+
+            if (partyTypesAllowed == null)
+            {
+                // if party types not set, all parties are allowed to initiate
+                return true;
+            }
+
+            if (partyTypesAllowed.BankruptcyEstate == false && partyTypesAllowed.Organization == false && partyTypesAllowed.Person == false && partyTypesAllowed.SubUnit == false)
```

Comment:
Check if there exists a shorthand for this notation ? 
Also it could be combined with the if check above to limit lines of code.

Reply:
Ah, good catch.